src/DatePicker/DatePicker.test.js

```diff
@@ -142,77 +141,53 @@ describe('<DatePicker />', () => {
     test('entering start date and disabled end range dates', () => {
         wrapper = mountComponentWithStyles(disabledFutureRangePicker);
         // set dates
-        let startRangeDate = new Date();
-        let endRangeDate = new Date();
-        endRangeDate.setDate(endRangeDate.getDate() + 3);
+        let startRangeDate = moment();
+        let endRangeDate = moment();
+        endRangeDate.add(3, 'day');
 
-        let startDate = `${startRangeDate.getMonth() +
-            1}/${startRangeDate.getDate()}/${startRangeDate.getFullYear()}`;
-        let endDate = `${endRangeDate.getMonth() +
-            1}/${endRangeDate.getDate()}/${endRangeDate.getFullYear()}`;
+        let startDate = `${startRangeDate.month() +
+            1}/${startRangeDate.date()}/${startRangeDate.year()}`;
+        let endDate = `${endRangeDate.month() +
+            1}/${endRangeDate.date()}/${endRangeDate.year()}`;
 
         wrapper.find('input[type="text"]')
             .simulate('change', { target: { value: `${startDate}-${endDate}` } });
 
         wrapper.find('input[type="text"]').simulate('keypress', { key: 'Enter' });
 
-        expect(wrapper.state('arrSelectedDates')).toEqual('undefined');
-    });
-
-    test('entering invalid range dates', () => {
```

Comment:
DatePicker used to arbitrarily set a limit of 3000 for the year. I do not think we need to impose this limitation now that moment is handling validation.